### PR TITLE
Update changelog 0.4.0

### PR DESCRIPTION
**Pull request by vince-fugnitto**
- Added new items for the release of `@theia:0.4.0`.
- Refactored existing items (fixed typos, formatting).

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,79 @@
...
-- [plugin] Introduce `vscode.previewHtml` command support
+- [application-manager] added support for pre-load HTML templates
+- [console] added support for console `when` contexts
+- [core] added support for os `when` contexts
+- [core] added support for shell `when` contexts
+- [core] added support for vscode closure contexts
+- [core] fixed bad vertical resizing behavior
+- [core] improve scrollbar visibility for the command palette
+- [core] improve tab-bar display (display 'X' (close) on dirty editors when hovering over dirty icon)
+- [core] improve tab-bar display (display 'X' (close) only when current editor is active, or has hover)
```
Comment (elaihau) [inline CHANGELOG.md]: one thing i am not sure about is the verb that starts each line. should we use current or past tense? they are mixed across the log.

Comment (vince-fugnitto) [reply CHANGELOG.md]: thanks @elaihau I must have missed some, generally we follow the past tense.

**Comment by marcdumais-work** [comment]
Thanks @vince-fugnitto . I directly included your version of the changelog in the release commit, to avoid having one more round of CI/publish before I could go ahead and do the release.

**Comment by vince-fugnitto** [comment]
> Thanks @vince-fugnitto . I directly included your version of the changelog in the release commit, to avoid having one more round of CI/publish before I could go ahead and do the release.

thanks @marcdumais !

**Pull request closed without merging**